hardware_interface/include/hardware_interface/handle.hpp

```diff
@@ -85,8 +87,9 @@ class Handle
     else
     {
       throw std::runtime_error(
-        "Invalid data type : '" + interface_description.interface_info.data_type +
-        "' for interface : " + interface_description.get_name());
+        fmt::format(
+          FMT_COMPILE("Invalid data type : '{}' for interface : {}"),
+          interface_description.interface_info.data_type, interface_description.get_name()));
     }
```

Comment:
@bmagyar I would like to know your opinion here. fmt is cool when we use it for the string creation etc. I'm just not sure for the exceptions. Do you have a strong opinion here?

Reply:
I quite like the use of format strings in general. I think it's still useful in exceptions